src/main.rs

```diff
@@ -426,13 +426,27 @@ async fn start_ldk() {
 	}
 
 	// Step 11: Optional: Initialize the NetGraphMsgHandler
-	// XXX persist routing data
 	let genesis = genesis_block(args.network).header.block_hash();
-	let router = Arc::new(NetGraphMsgHandler::new(
-		genesis,
+	let network_graph_path = format!("{}/network_graph", ldk_data_dir.clone());
+	let network_graph = disk::read_network(Path::new(&network_graph_path), genesis);
+	let router = Arc::new(NetGraphMsgHandler::from_net_graph(
 		None::<Arc<dyn chain::Access + Send + Sync>>,
 		logger.clone(),
+		network_graph,
 	));
+	let router_persist = Arc::clone(&router);
+	tokio::spawn(async move {
+		let mut interval = tokio::time::interval(Duration::from_secs(60));
+		loop {
+			interval.tick().await;
+			// We ignore persistence errors here. They are non-fatal as we can just fetch the
```

Comment:
log warning maybe? could affect UX but be non-obvious

Reply:
Done!